eventhubs/data-plane/azure-eventhubs-eph/src/main/java/com/microsoft/azure/eventprocessorhost/InMemoryLeaseManager.java

```diff
@@ -403,5 +406,19 @@ public boolean isExpiredSync() {
             TRACE_LOGGER.debug("isExpired(" + this.getPartitionId() + (hasExpired ? ") expired " : ") leased ") + (this.expirationTimeMillis - System.currentTimeMillis()));
             return hasExpired;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            if (!super.equals(o)) return false;
+            InMemoryLease that = (InMemoryLease) o;
+            return expirationTimeMillis == that.expirationTimeMillis;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(super.hashCode(), expirationTimeMillis);
+        }
```

Comment:
The equals() and hashCode() are for the InMemoryLease but not InMemoryLeaseStore. The expirationTimeMillis value is kind of make sense to be included in the equals() method. I think it is the expiration time for the InMemoryLease. If that is incorrect, since there is only variable difference to InMemoryLease's parent class CompleteLease. I would just call super.equals() and super.hashCode() in the InMemoryLease's equals and hashCode method?

Reply:
I'm not clear why these methods are required? This is a singleton class - it should never be equal to anything, as there should only ever be the one instance.

Reply:
InMemory**LeaseStore** is singleton but InMemory**Lease** is not. InMemoryLease can be store into the ConcurrentHashMap<String, InMemoryLease> inMemoryLeasesPrivate;. Since the Map interface is involved, we should add the equals() and hashCode() to InMemoryLesse class

Reply:
If it is being stored in the Map as the value, not the key, then the equals / hashCode methods will never be relevant. Maps only use the equals / hashCode methods on the key. However, I think it is better to resolve this issue as you have it above, rather than suppress it in spotbugs (as that becomes too noisy).

Reply:
Thank you for the correction. So we leave the equals() with expirationTimeMillis as it is.